src/python/pants/task/goal_options_mixin.py

```diff
@@ -0,0 +1,34 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.option.optionable import Optionable
+from pants.option.scope import ScopeInfo
+
+
+class GoalOptionsRegistrar(Optionable):
+  """Subclass this to register recursive options on all tasks in a goal.
+
+  This is useful when you want the possibility of setting the value for all tasks
```

Comment:
It seems like offering the ability to set an option for all tasks in a goal via the goal's scope ought to fail if not all tasks installed in the goal support the option.

Reply:
Not sure how you would test for that. For example, to inherit `--skip` the task code needn't change at all, as the use of `--skip` is implemented in the engine. In any case, it seems no worse than registering an option on some task but having that task's code not use the option, which isn't something we can guard against anyway.

Reply:
Yeah. I was thinking of it from the user expectation side. `./pants compile --lint-skip` for a lint goal with a lint task installed that did not mixin skip would be confusing to the user since one of the lints would still run. This sort of confusion can't happen, except by coding error as you point out, for an individual task skip option - which presumably is normally actually implemented.

Reply:
Yes, in new engine land that makes sense. But all tasks will be written from scratch for that, so we have time to consider what goal-level options should look like in that case... In fact, I'm not even sure what goals are in the new engine.